concent_api/command_line_tool/message_extractor.py

```diff
@@ -0,0 +1,122 @@
+from base64 import b64decode
+import importlib
+import re
+import random
+import string
+from typing import Dict, Text, Any, List, Union
+from golem_messages.datastructures import FrozenDict
+from golem_messages.message import Message
+
+JsonType = Dict[Text, Any]
+Task = FrozenDict
+
+
+def make_random_string(length=None, chars=None):
+    length = length if length is not None else 36
+    chars = chars if chars is not None else (string.ascii_letters + string.digits)
+
+    return ''.join(random.choice(chars) for _ in range(length))
+
+
+def split_uppercase(message):
+    return (''.join([(word[:1].lower() + word[1:]) for word in re.sub(r'([A-Z])', r'_\1', message)]))[1:]
+
+
+def find_modules():
+    message_tasks = importlib.import_module("golem_messages.message.tasks")
+    message_concents = importlib.import_module("golem_messages.message.concents")
+    message_modules = [message_tasks, message_concents]
+    message_list = []
+    for module in message_modules:
+        (message_list.extend([name for name, cls in module.__dict__.items() if isinstance(cls, type)]))
+    return message_list
+
+
+def get_field_names():
+    field_names = []
+    for message in find_modules():
+        field_names.append(split_uppercase(message))
+    return field_names
+
+
+FIELD_NAMES = get_field_names()
+
+
+def validate_message_list(message_list: List[Message]) -> None:
+    if len(message_list) > 1:
+        raise ValueError("Malformed message definition")
+
+
+def create_message(message_name: str, message_params: JsonType) -> Union[Message, Task]:
+    module = importlib.import_module("golem_messages.message")
+    module = module if hasattr(module, convert_message_name(message_name)) else importlib.import_module(
+        "golem_messages.message.concents")
+    message = getattr(module, convert_message_name(message_name))(**message_params)
+    return message
+
+
+def substitute_message(json: JsonType, message_name: str, message: Message) -> JsonType:
+    params = {k: v for k, v in json.items()}
+    params[message_name] = message
+    return params
+
+
+def convert_message_name(message):
+    return ''.join([(word[:1].capitalize() + word[1:]) for word in message.split('_')])
+
+
+def _contains_valid_message(messages: List[Text]) -> bool:
```

Comment:
The name is not exactly saying what this function does. It suggest it does some kind of validation, but it only checks if list length is equal 1. Please rename.

Reply:
Done.